Makefile

```diff
@@ -92,133 +93,147 @@ deploy-live: prod
 
 # Tests
 
-# set a default test command for convenience
-test: test-contracts
+# set a default test command for developer convenience
+test: test-default
+test-default: test-client
+test-all: test-client test-contracts test-hub test-integration
 
-test-contracts: client contract-artifacts
-	bash $(contracts)/ops/test.sh
+test-contracts: contract-artifacts
+	bash ops/test-contracts.sh
 
-test-hub: client hub-node-modules
-	echo coming soon!
+test-hub: hub database ethprovider
+	bash ops/test-hub.sh
 
-test-client: client hub-node-modules
-	echo coming soon!
+test-client: client
+	bash ops/test-client.sh
 
-test-integration: dev
-	echo coming soon!
+test-e2e: root-node-modules prod
+	npm stop
+	npm run prod
+	./node_modules/.bin/cypress run
+	npm stop
 
+########################################
 # Begin Real Rules
 
 # Proxy
 
 proxy-prod: wallet-prod $(shell find $(proxy) $(find_options))
-	$(log_start)
+	$(log_start) && echo "prereqs: $<"
 	docker build --file $(proxy)/prod.dockerfile --tag $(project)_proxy:latest .
 	$(log_finish) && touch build/proxy-prod
 
 proxy: $(shell find $(proxy) $(find_options))
-	$(log_start)
+	$(log_start) && echo "prereqs: $<"
 	docker build --file $(proxy)/dev.dockerfile --tag $(project)_proxy:dev .
 	$(log_finish) && touch build/proxy
 
 # Wallet
 
 wallet-prod: wallet-node-modules $(shell find $(wallet)/src $(find_options))
-	$(log_start)
+	$(log_start) && echo "prereqs: $<"
 	$(docker_run_in_wallet) "rm -f .env && cp ops/prod.env .env"
 	$(docker_run_in_wallet) "npm run build"
 	$(log_finish) && touch build/wallet-prod
 
 wallet: wallet-node-modules $(shell find $(wallet)/src $(find_options))
-	$(log_start)
+	$(log_start) && echo "prereqs: $<"
 	docker build --file $(wallet)/ops/dev.dockerfile --tag $(project)_wallet:dev $(wallet)
 	$(log_finish) && touch build/wallet
 
 wallet-node-modules: builder client $(wallet)/package.json
-	$(log_start)
+	$(log_start) && echo "prereqs: $<"
 	$(docker_run_in_wallet) "rm -rf node_modules/connext"
-	$(docker_run_in_wallet) "$(install)"
+	$(docker_run_in_wallet) "$(install)" > /dev/null
```

Comment:
install output spams the CI logs so this quiets things but still allows errors & warning to be printed